components/brave_rewards/resources/shared/types/index.ts

```diff
@@ -0,0 +1,340 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import { ConnectExternalWalletError, WalletStatus } from 'gen/brave/components/brave_rewards/common/mojom/ledger_types.mojom.m.js'
```

Comment:
If this file is only used by the stuff in `page` then it can go in `page/lib/` instead of `shared`.

Reply:
So maybe `components/brave_rewards/resources/page/lib/types.ts`?

Reply:
:+1: good call! I'm wondering if we should add a comment saying you don't have to add your new types in the file? Personally, I prefer my types to live next to where they're used, but I don't know what rewards would like?